homeassistant/components/group/entity.py

```diff
@@ -368,9 +367,24 @@ def async_update_group_state(self) -> None:
         self._state = None
         self._async_update_group_state()
 
+    @callback
+    def async_domain_registered(self, domain: str) -> None:
+        """Refresh after a new entity platform has been registered."""
+        _LOGGER.debug(
+            "Domain %s was registered, refreshing state %s",
```

Comment:
The downside here is this quadratic in respect to the number of domains registered and groups.  Ideally we only write state if it's a domain we care about. 

We could only receive signals for domains we care about as well if we had a signal per domain but that might get a bit complex because we would have to disconnect or connect dispatchers if the domains being grouped changes